qiskit/pulse/timeslots.py

```diff
@@ -370,8 +370,12 @@ def shift(self, time: int) -> 'TimeslotCollection':
         Args:
             time: time to be shifted by
         """
-        slots = [slot.shift(time) for slot in self.timeslots]
-        return TimeslotCollection(*slots)
+        new_tc = TimeslotCollection()
+        new_tc_table = new_tc._table
+        old_td_table = self._table
+        for channel, timeslots in old_td_table.items():
```

Comment:
I think this is more readable without the local var here, as just:

```suggestion
        for channel, timeslots in self._table.items():
```

Reply:
Fixed in e10c74c8